ddl/index.go

```diff
@@ -709,6 +706,7 @@ func onDropIndex(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	default:
 		err = dbterror.ErrInvalidDDLState.GenWithStackByArgs("index", indexInfo.State)
```

Comment:
If we do this change, do we need to do a return here? Otherwise, there is an error, but also change the schema version?

Reply:
I think this is fine because the DDL state is incorrect anyway.

`job.SchemaState` is exactly used to show the `indexInfo.State` here.

Reply:
However, this may cause a problem if it is later written to KV (DDL job should be retried).